hail/src/main/scala/is/hail/stats/PearsonCorrelationCombiner.scala

```diff
@@ -0,0 +1,63 @@
+package is.hail.stats
+
+import is.hail.annotations.{Annotation, RegionValueBuilder}
+
+class PearsonCorrelationCombiner extends Serializable {
+  // formula from https://en.wikipedia.org/wiki/Pearson_correlation_coefficient
+
+  var xSum: Double = 0d
+  var ySum: Double = 0d
+  var xSqSum: Double = 0d
+  var ySqSum: Double = 0d
+  var xySum: Double = 0d
+  var n: Long = 0L
+
+  def merge(x: Double, y: Double) {
+    xSum += x
+    ySum += y
+    xSqSum += x * x
+    ySqSum += y * y
+    xySum += x * y
+    n += 1
+  }
+
+  def merge(other: PearsonCorrelationCombiner) {
+    xSum += other.xSum
+    ySum += other.ySum
+    xSqSum += other.xSqSum
+    ySqSum += other.ySqSum
+    xySum += other.xySum
+    n += other.n
+  }
+
+  private def computeResult(): Double = (n * xySum - xSum * ySum) / (math.sqrt(n * xSqSum - xSum * xSum) * math.sqrt(n * ySqSum - ySum * ySum))
```

Comment:
replace denominator with `math.sqrt((n * xSqSum - xSum * xSum) * (n * ySqSum - ySum * ySum))` to save a sqrt

Reply:
done